hera_cal/redcal.py

```diff
@@ -1957,6 +1963,123 @@ def redcal_run(input_data, filetype='uvh5', firstcal_ext='.first.calfits', omnic
     return cal
 
 
+def nightly_median_firstcal_delays(redcal_meta_file_list, output_ext='.redcal_meta.median_phases.hdf5',
+                                   output_replace='.redcal_meta.hdf5'):
+    """
+    Find the median delay and polarity for list of firstcal meta files.
+    and write them out with that median.
+
+    Parameters
+    ----------
+    redcal_meta_file_list: list of str
+        list of file names containing redcal meta data. All files should have same number of times.
+
+    """
+    redcal_metas = []
+    for meta_file in redcal_meta_file_list:
+        redcal_metas.append((meta_file, ) + read_redcal_meta(meta_file))
+    # build list of delays and polarity flips for each antenna.
+    nfiles = len(redcal_meta_file_list)
+    ntimes = [len(rcm[4]) for rcm in redcal_metas]
+    delays = {ant: np.zeros(np.sum(ntimes)) for ant in redcal_metas[0][1]['dlys']}
+    offsets = {ant: np.zeros(np.sum(ntimes)) for ant in redcal_metas[0][1]['offsets']}
+    polarity_flips = {ant: copy.deepcopy(delays[ant]).astype(bool) for ant in delays}
+    nt = 0
+    for filenum, (meta_filename, fc_meta, omni_meta, freqs, times, lsts, antpos, history) in enumerate(redcal_metas):
+        for ant in fc_meta['dlys']:
+            tslice = slice(nt, nt + len(times))
+            delays[ant][tslice] = fc_meta['dlys'][ant]
+            polarity_flips[ant][tslice] = fc_meta['polarity_flips'][ant]
+            offsets[ant][tslice] = fc_meta['offsets'][ant]
+        nt += len(times)
+    # compute medians.
+    for ant in delays:
+        delays[ant] = np.nanmedian(delays[ant])
+        polarity_flips[ant] = bool(np.nanmedian(polarity_flips[ant]))
```

Comment:
If all the polarity flips are `np.nan` for a given antenna, this will turn into `True` though I think you want it to be `False` by default. Probably doesn't matter but you might want an a check on `not np.any(np.isfinite())`